fv3core/tests/main/test_dycore_call.py

```diff
@@ -0,0 +1,201 @@
+import contextlib
+import os
+import unittest.mock
+from typing import Any, List, Tuple
+
+import fv3core
+import fv3core._config
+import fv3core.initialization.baroclinic as baroclinic_init
+import fv3core.testing
+import pace.util
+from fv3core.grid import MetricTerms
+from fv3core.utils.grid import DampingCoefficients, GridData
+from fv3core.utils.null_comm import NullComm
+
+
+DIR = os.path.abspath(os.path.dirname(__file__))
+
+
+@contextlib.contextmanager
+def no_lagrangian_contributions(dynamical_core: fv3core.DynamicalCore):
+    # TODO: lagrangian contributions currently cause an out of bounds iteration
+    # when halo updates are disabled. Fix that bug and remove this decorator.
+    # Probably requires an update to gt4py (currently v36).
+    def do_nothing(*args, **kwargs):
+        pass
+
+    original_attributes = {}
+    for obj in (
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_delz,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_pt,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_u,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_v,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_w,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_delz,
+    ):
+        original_attributes[obj] = obj._lagrangian_contributions
+        obj._lagrangian_contributions = do_nothing  # type: ignore
+    for (
+        obj
+    ) in dynamical_core._lagrangian_to_eulerian_obj._mapn_tracer._list_of_remap_objects:
+        original_attributes[obj] = obj._lagrangian_contributions
+        obj._lagrangian_contributions = do_nothing  # type: ignore
+    try:
+        yield
+    finally:
+        for obj, original in original_attributes.items():
+            obj._lagrangian_contributions = original
+
+
+def setup_dycore() -> Tuple[fv3core.DynamicalCore, List[Any]]:
+    backend = "numpy"
+    stencil_config = fv3core.StencilConfig(
+        backend=backend,
+        rebuild=False,
+        validate_args=True,
+    )
+    mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
```

Comment:
if you define this after the config, you could use the layout to determine the total_ranks. not a required change since it is test code

Reply:
Good idea, in the end I'll avoid doing this only because in practice we'll want to either configure layout from the MPI comm or validate layout from the MPI comm, and this example might confuse someone into writing it the other way around.

Reply:
Wait my comment makes no sense, in this production example it's not possible to make an MPI comm! So I'll put this in.